### PR TITLE
[Refactor] Relay Node::make to constructor

### PR DESCRIPTION
**Pull request by zhiics**
#5069

### CODE REVIEW

File: include/tvm/relay/adt.h [hunk 1]
```diff
@@ -91,13 +91,9 @@ class PatternVar;
 /*! \brief PatternVar container node */
```
Comment (zhiics) [inline include/tvm/relay/adt.h]: @tqchen Do we want to move `make` in `PatternWildcardNode` to `PatternWildcard`? There is already a default constructor for `PatternWildcard`.

File: src/tir/ir/data_layout.cc [hunk 2]
```diff
@@ -350,20 +350,16 @@ Array<PrimExpr> BijectiveLayout::BackwardShape(const Array<PrimExpr>& shape) con
...
 
-  n->src_layout = src_layout;
-  n->dst_layout = dst_layout;
+  n->src_layout = std::move(src_layout);
+  n->dst_layout = std::move(dst_layout);
 
-  if (!GetStoreRule(&n->forward_rule, n->src_layout, n->dst_layout)) {
-    // not convertible
-    return BijectiveLayout();
+  if (GetStoreRule(&n->forward_rule, n->src_layout, n->dst_layout)) {
```
Comment (zhiics) [inline src/tir/ir/data_layout.cc]: @yzhliu @anijain2305 please take a look here

File: include/tvm/relay/adt.h (hunk 1 above)
Comment (tqchen) [reply include/tvm/relay/adt.h]: Yes, We can allow multiple constructors as long as there is no ambiguity

Comment (zhiics) [reply include/tvm/relay/adt.h]: This one would be ambiguous if we create another constructor `PatternWildcard()`

Comment (tqchen) [reply include/tvm/relay/adt.h]: I see, in this case we should simply override the default constructor to the new behavior(do not use the original constructor). We would likely need to add copy/move constructor and operator= overloads though in this case as well.

File: include/tvm/relay/adt.h [hunk 3]
```diff
@@ -28,9 +28,10 @@
 #include <tvm/ir/adt.h>
 #include <string>
 #include <functional>
-#include "./base.h"
-#include "./type.h"
-#include "./expr.h"
+#include <utility>
+#include "base.h"
```
Comment (tqchen) [inline include/tvm/relay/adt.h]: To be consistent with the rest of the code, ` #include <tvm/relay/base.h>`

File: include/tvm/relay/adt.h (hunk 1 above)
Comment (zhiics) [reply include/tvm/relay/adt.h]: done. PTAL

File: src/tir/ir/data_layout.cc (hunk 2 above)
Comment (tqchen) [reply src/tir/ir/data_layout.cc]: add a comment: to be consistent with old behavior, create a nullptr layout when argument is invalid

**Comment by tqchen** [comment]
@zhiics would you be able to do a quick patch to fix the remaining issue on the master?

**Comment by zhiics** [comment]
I can spend some time on it late today? could you please give a pointer to the remaining issue?

**Comment by tqchen** [comment]
Seems was a false alarm, neve rmind

**Pull request merged**